File: Java/src/main/java/Solution201_300/Solution258.java
package Solution201_300;

/**
 * @author windows
 */
public class Solution258 {

    public int addDigits(int num) {
        while(num >= 10){
            int temp = num;
            int sum = 0;
            while (temp > 0){
                sum += temp % 10;
                temp /= 10;
            }
            num = sum;
        }
        return num;
    }


    public static void main(String[] args) {
        Solution258 solution258 = new Solution258();
    }
}

File: C++/src/Solution201_300/Solution258.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution258.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int addDigits(int num) {
  while (num >= 10){
    int n = num;
    int sum = 0;
    while (n > 0){
      sum += n % 10;
      n /= 10;
    }
    num = sum;
  }
  return num;
}